Source/Core/EllipsoidRhumbLine.js

```diff
@@ -0,0 +1,431 @@
+        var cos4D = Math.cos(4 * d);
+        var sin6D = Math.sin(6 * d);
+        var cos6D = Math.cos(6 * d);
+        var sin8D = Math.sin(8 * d);
+        var cos8D = Math.cos(8 * d);
+        var sin10D = Math.sin(10 * d);
+        var cos10D = Math.cos(10 * d);
+        var sin12D = Math.sin(12 * d);
+
+        return d + d * e2 / 4 + 7 * d * e4 / 64 + 15 * d * e6 / 256 + 579 * d * e8 / 16384 + 1515 * d * e10 / 65536 + 16837 * d * e12 / 1048576
+            + (3 * d * e4 / 16 + 45 * d * e6 / 256 - d * (32 * d2 - 561) * e8 / 4096 - d * (232 * d2 - 1677) * e10 / 16384 + d * (399985 - 90560 * d2 + 512 * d4) * e12 / 5242880) * cos2D
+            + (21 * d * e6 / 256 + 483 * d * e8 / 4096 - d * (224 * d2 - 1969) * e10 / 16384 - d * (33152 * d2 - 112599) * e12 / 1048576) * cos4D
+            + (151 * d * e8 / 4096 + 4681 * d * e10 / 65536 + 1479 * d * e12 / 16384 - 453 * d3 * e12 / 32768) * cos6D
+            + (1097 * d * e10 / 65536 + 42783 * d * e12 / 1048576) * cos8D
+            + 8011 * d * e12 / 1048576 * cos10D
+            + (3 * e2 / 8 + 3 * e4 / 16 + 213 * e6 / 2048 - 3 * d2 * e6 / 64 + 255 * e8 / 4096 - 33 * d2 * e8 / 512 + 20861 * e10 / 524288 - 33 * d2 * e10 / 512 + d4 * e10 / 1024 + 28273 * e12 / 1048576 - 471 * d2 * e12 / 8192 + 9 * d4 * e12 / 4096) * sin2D
+            + (21 * e4 / 256 + 21 * e6 / 256 + 533 * e8 / 8192 - 21 * d2 * e8 / 512 + 197 * e10 / 4096 - 315 * d2 * e10 / 4096 + 584039 * e12 / 16777216 - 12517 * d2 * e12 / 131072 + 7 * d4 * e12 / 2048) * sin4D
+            + (151 * e6 / 6144 + 151 * e8 / 4096 + 5019 * e10 / 131072 - 453 * d2 * e10 / 16384 + 26965 * e12 / 786432 - 8607 * d2 * e12 / 131072) * sin6D
+            + (1097 * e8 / 131072 + 1097 * e10 / 65536 + 225797 * e12 / 10485760 - 1097 * d2 * e12 / 65536) * sin8D
+            + (8011 * e10 / 2621440 + 8011 * e12 / 1048576) * sin10D
+            + 293393 * e12 / 251658240 * sin12D;
+    }
+
+    function calculateSigma(ellipticity, latitude) {
+        if (ellipticity === 0.0) { // sphere
+            return Math.log(Math.tan(0.5 * (CesiumMath.PI_OVER_TWO + latitude)));
+        }
+
+        var eSinL = ellipticity * Math.sin(latitude);
+        return Math.log(Math.tan(0.5 * (CesiumMath.PI_OVER_TWO + latitude))) - (ellipticity / 2.0 * Math.log((1 + eSinL) / (1 - eSinL)));
+    }
+
+    function calculateHeading(ellipsoidRhumbLine, major, minor, firstLongitude, firstLatitude, secondLongitude, secondLatitude) {
+        var sigma1 = calculateSigma(ellipsoidRhumbLine._ellipticity, firstLatitude);
+        var sigma2 = calculateSigma(ellipsoidRhumbLine._ellipticity, secondLatitude);
+        return Math.atan2(CesiumMath.negativePiToPi(secondLongitude - firstLongitude), sigma2 - sigma1);
+    }
+
+    function calculateArcLength(ellipsoidRhumbLine, major, minor, firstLongitude, firstLatitude, secondLongitude, secondLatitude) {
+        //>>includeStart('debug', pragmas.debug);
+        Check.defined('heading', ellipsoidRhumbLine._heading);
```

Comment:
Based on the callee it looks like there shouldn't be a situation where _heading doesn't exist?